scipy/stats/tests/test_stats.py

```diff
@@ -3711,8 +3711,8 @@ def test_1samp_ci_iv(self, xp):
     def test_pvalue_ci(self, alpha, data_axis, alternative, xp):
         # test relationship between one-sided p-values and confidence intervals
         data, axis = data_axis
-        data = data.astype(copy=True)  # ensure byte order
-        data = xp.asarray(data)
+        data = data.astype(np.float64, copy=True)  # ensure byte order
+        data = xp.asarray(data, dtype=xp.float64)
```

Comment:
```suggestion
        data = data.astype(np.float64, copy=True)  # ensure byte order
        # `float64` is needed here because of the range of `alpha`
        data = xp.asarray(data, dtype=xp.float64)
```

Reply:
I did check that all these incantations pass locally on x86_64 Linux, while they fail on `main`:

```
SCIPY_XSLOW=1 SCIPY_DEVICE=cuda python dev.py test -j 32 -b all -- -k "test_pvalue_ci"
SCIPY_XSLOW=1 python dev.py test -j 32 -b all -- -k "test_pvalue_ci"
SCIPY_XSLOW=1 python dev.py test -j 32 -- -k "test_pvalue_ci"
```

I tried to play picky reviewer and force `float32` instead of `float64` locally, and saw overflow/invalid value warnings, so that seems like a plausible justification for non-`stats` expert at least.

Reply:
(obivously the first one has limited value, since it is cpu only)